### DISCUSSION TITLE
Variables not set. Defaulting to blank string

### DISCUSSION THREAD

**Discussion by Nevrai** [danny-avila/LibreChat#4235] [Troubleshooting] [answered]
### What happened?

I installed LibreChat nine days ago. I run it with the Docker desktop program. I can open the local LibreChat application via Docker Desktop, but for some reason, I can’t use the `compose up` or `compose down` commands anymore. I also want to update LibreChat, but `git pull` isn’t working. I haven’t modified anything in the LibreChat files, so I don’t know why it’s not working.

`docker compose up`:
```
time="2024-09-24T22:52:55+02:00" level=warning msg="The \"PORT\" variable is not set. Defaulting to a blank string."
time="2024-09-24T22:52:55+02:00" level=warning msg="The \"PORT\" variable is not set. Defaulting to a blank string."
time="2024-09-24T22:52:55+02:00" level=warning msg="The \"UID\" variable is not set. Defaulting to a blank string."
time="2024-09-24T22:52:55+02:00" level=warning msg="The \"GID\" variable is not set. Defaulting to a blank string."
time="2024-09-24T22:52:55+02:00" level=warning msg="The \"UID\" variable is not set. Defaulting to a blank string."
time="2024-09-24T22:52:55+02:00" level=warning msg="The \"GID\" variable is not set. Defaulting to a blank string."
time="2024-09-24T22:52:55+02:00" level=warning msg="The \"UID\" variable is not set. Defaulting to a blank string."
time="2024-09-24T22:52:55+02:00" level=warning msg="The \"GID\" variable is not set. Defaulting to a blank string."
time="2024-09-24T22:52:55+02:00" level=warning msg="C:\\Users\\Greg\\Desktop\\Greg Stuff\\Programs\\LibreChat-main\\docker-compose.yml: the attribute `version` is obsolete, it will be ignored, please remove it to avoid potential confusion"
no port specified: :<empty>
```

`git pull`:
`fatal: not a git repository (or any of the parent directories): .git

***

**OS**: Windows 11 Home

### Steps to Reproduce

Use the specified commands in the console.

Install folder:
![image](https://github.com/user-attachments/assets/95c84883-a811-4a8b-9c96-f145a2e905d8)

I run a command prompt from here.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
1. First, it looks like you might not be in the correct directory when running your commands. Make sure you're in the LibreChat folder before running any commands. You can do this by typing:

   ```
   cd C:\Users\Greg\Desktop\Greg Stuff\Programs\LibreChat-main
   ```

(side note: are you sure `LibreChat-main` is the correct name of the folder? make sure this matches exactly)

2. The 'git pull' error suggests that the directory isn't recognized as a git repository. This could be because you're not in the right folder, or the .git folder is missing. After changing to the correct directory, check if the .git folder exists.

3. For the Docker Compose warnings, you can ignore them, as they are not really needed in a windows environment.

4. Since you can run LibreChat through Docker Desktop, you could try using Docker Desktop's terminal, which should already be in the correct context.

5. If git commands still aren't working, you might need to re-clone the repository. Here's how:

   ```
   cd C:\Users\Greg\Desktop\Greg Stuff\Programs\
   rename LibreChat-main LibreChat-old
   git clone https://github.com/danny-avila/LibreChat.git LibreChat-main
   ```

   After cloning, copy your .env file from the old directory to the new one.

Please try these steps and let me know if you're still having issues. Remember to always backup your data and configuration before making significant changes!

If you're still stuck after trying these suggestions, please provide:
1. The output of `pwd` (or `cd` on Windows) to confirm your current directory
2. The contents of your .env file (excluding any sensitive information)
3. The result of `ls -la` (or `dir` on Windows) in your LibreChat directory

This additional information will help us troubleshoot further if needed. Good luck!

**Comment by Nevrai** [comment, marked as answer]
Thank you for the thorough answer.

It turns out that I am absolutely moronic. I was running the commands from the original installation folder that wasn’t being used, rather than from the actual folder (which was `%userprofile%\LibreChat`). My apologies for wasting your time.
